src/stories/StyledTab.tsx

```diff
@@ -0,0 +1,79 @@
+import { PropsWithChildren, ReactElement } from 'react';
+import DfxIcon, { IconColors, IconSizes, IconVariant } from './DfxIcon';
+import { TabType } from './StyledTabContainer';
+
+interface StyledTabProps extends PropsWithChildren {
+  tabId: number;
+  updateActiveTab: (tId: number) => any;
+  openTab: number;
+  deactivated?: boolean;
+  type?: TabType;
+}
+
+const TABTYPE_MAP: Record<TabType, ReactElement> = {
+  [TabType.BUY]: <IconTab icon={IconVariant.BANK} />,
+  [TabType.SOON]: <ComingSoon />,
```

Comment:
I think this should be generic and be exposed as parameter. Maybe export the `IconTab` component?

Reply:
did a re-write of this functionality. now the icon or flagwords can be passed down and the visual-addon is generated according to the case ...